shell/plugins/steve/subscribe.js

```diff
@@ -579,15 +589,46 @@ export const actions = {
 
     // console.warn(`Resource stop: [${ getters.storeName }]`, msg); // eslint-disable-line no-console
 
+    // If we're trying to watch this event, attempt to re-watch
     if ( getters['schemaFor'](type) && getters['watchStarted'](obj) ) {
       // Try reconnecting once
 
       commit('setWatchStopped', obj);
 
+      // In summary, we need to re-watch but with a reliable `revision` (to avoid `too old` message kicking off a full re-fetch of all
+      // resources). To get a reliable `revision` go out and fetch the latest for that resource type, in theory our local cache should be
+      // up to date with that revision.
+      // Optimisation - Some v1 resource types don't have revisions (either at the collection or resource level), so avoid trying to fetch
+      // latest for those types
+      // Note - In theory `0`, `-1` or `null` revisions will watch for latest, however steve will send the current state of each resource
+      // via a `resource.created` event
+
+      const revision = getters.nextResourceVersion(type, obj.id);
+
+      let pLatestRevision;
```

Comment:
nit: what is the "p" in "pLatestRevision" here?

Reply:
Old habit, it denotes the type of variable (promise). Think i originally had it `revision` and `pRevision`. Will update